x/bridge/keeper/assets_unlocked.go

```diff
@@ -0,0 +1,117 @@
+package keeper
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/common"
+
+	"cosmossdk.io/math"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/mezo-org/mezod/x/bridge/types"
+	evmtypes "github.com/mezo-org/mezod/x/evm/types"
+)
+
+// GetAssetsUnlockedSequenceTip returns the current sequence tip for the
+// AssetsUnlocked events. The tip denotes the sequence number of the last event
+// processed by the x/bridge module.
+func (k Keeper) GetAssetsUnlockedSequenceTip(ctx sdk.Context) math.Int {
+	bz := ctx.KVStore(k.storeKey).Get(types.AssetsUnlockedSequenceTipKey)
+
+	var sequenceTip math.Int
+	err := sequenceTip.Unmarshal(bz)
+	if err != nil {
+		panic(err)
+	}
+
+	if sequenceTip.IsNil() {
+		sequenceTip = math.ZeroInt()
+	}
+
+	return sequenceTip
+}
+
+// SetAssetsUnlockedSequenceTip sets the current sequence tip for the AssetsUnlocked
+// events. The tip denotes the sequence number of the last event processed by
+// the x/bridge module.
+func (k Keeper) setAssetsUnlockedSequenceTip(
+	ctx sdk.Context,
+	sequenceTip math.Int,
+) {
+	bz, err := sequenceTip.Marshal()
+	if err != nil {
+		panic(err)
+	}
+
+	ctx.KVStore(k.storeKey).Set(types.AssetsUnlockedSequenceTipKey, bz)
+}
+
+func (k Keeper) saveAssetsUnlocked(
+	ctx sdk.Context,
+	assetsUnlocked *types.AssetsUnlockedEvent,
+) {
+	bz, err := assetsUnlocked.Marshal()
+	if err != nil {
+		panic(err)
+	}
+
+	ctx.KVStore(k.storeKey).Set(types.GetAssetsUnlockedKey(assetsUnlocked.UnlockSequence), bz)
+}
+
+func (k Keeper) GetAssetsUnlocked(
+	ctx sdk.Context,
+	sequence math.Int,
+) *types.AssetsUnlockedEvent {
+	bz := ctx.KVStore(k.storeKey).Get(types.GetAssetsUnlockedKey(sequence))
+
+	var assetsUnlocked types.AssetsUnlockedEvent
+	err := assetsUnlocked.Unmarshal(bz)
+	if err != nil {
+		panic(err)
+	}
+
+	return &assetsUnlocked
+}
+
+func (k Keeper) AssetsUnlocked(
+	ctx sdk.Context,
+	token []byte,
+	amount math.Int,
+	chain uint8,
+	recipient []byte,
+) (*types.AssetsUnlockedEvent, error) {
+	var targetToken common.Address
```

Comment:
Let's ensure `x/bridge` module doesn't directly depend on `go-ethereum`. Ideally, only `x/evm` and `precompile` packages should. This helps isolating the `go-ethereum` dependency. There are some utils in `x/evm` that `x/bridge` can use to deal with addresses: https://github.com/mezo-org/mezod/blob/cf4a4bc8ffcf9e7c46826c236b26b211921efcb5/x/evm/types/utils.go#L122. Let's leverage that and try to keep the depenendcy graph as `x/bridge` -> `x/evm` -> `go-ethereum`